src/backend/expungeservice/expunger/analyze.py

```diff
@@ -118,3 +134,166 @@ def __init__(self, name, dob, cases):
+        self.statute = statute
+        self.date = date
+
+    def __str__(self):
+        return ' '.join([str(self.code), str(self.checks), str(self.statute), str(self.date)])
+
+class ResultInElig_137_225_5(Result):
+    def __init__(self, **kwargs):
+        Result.__init__(self, code=ResultCode.INELIGIBLE,
+                        statute=Statute(137, 225, 5), **kwargs)
+
+class RecordAnalyzer(object):
+    """
+    A class for analyzing client's records
+
+    Attributes:
+        client: A Client instance
+    """
+
+    def __init__(self, client):
+        self.client = client
+
+    def _is_charge_level(charge, type_, class_):
+        check = 'Is the charge a {}'.format(type_)
+        if class_:
+            check += ' class {}'.format(class_)
+
+        result = (charge.level.type_ ==  type_ and
+                  (not class_ or charge.level.class_ == class_))
+        return Check(check=check, result=result)
+
+    def _is_charge_statute(charge, statute):
+        check = 'Does the charge fall under statute: ' + str(statute)
+        # TODO implement this
+        return Check(check=check, result=True)
+
+    def _is_charge_in_statute_list(charge, statutes, desc):
+        check = ('Does the charge fall under any of these statutes: ' +
+                 ','.join(str(statute) for statute in statutes))
+        # TODO implement this
+        return Check(check=check, result=False, check_desc=desc)
+
+    def _is_charge_sex_crime(charge):
+        # TODO update
+        _statutes_sex_crimes = []
+        return RecordAnalyzer._is_charge_in_statute_list(
+            charge, _statutes_sex_crimes, 'Is the charge a sex crime')
+
+    def _is_charge_traffic_crime(charge):
+        # TODO update
+        _statutes_traffic_crimes = []
+        return RecordAnalyzer._is_charge_in_statute_list(
+            charge, _statutes_traffic_crimes, 'Is the charge a traffic crime')
+
+    def _have_open_case(self):
+        check = 'Is there a open case for the client'
+        result = any([case.state == CaseState.OPEN for case in self.client.cases])
+        return Check(check=check, result=result)
+
+    """
+    Run Time Eligibility analysis on the client and their records
+
+    TODO: make it return analysis for each charge as well (which is supposed
+    to be an update on eligibility date and relevan statutes)
+
+    Returns:
+        An Result instance
+    """
+    def time_eligibility(self):
+        checks = []
+
+        checks.append(self._have_open_case())
+        if checks[-1].result:
+            return Result(ResultCode.OPEN_CASE, checks)
+
+        # TODO implement the rest
+
+        return Result(ResultCode.NO_ACTION)
+
+    """
+    Run Type Eligibility analysis on a charge
+
+    Args:
+        charge: A Charge instance
+
+    Returns:
+        A Result instance
+    """
+    def type_eligibility(self, charge):
+        checks = []
+
+        checks.append(RecordAnalyzer._is_charge_level(charge, 'Felony', 'A'))
```

Comment:
Do you think there is a better way to implement this then to go down a list and test for each charge in a given order?

Reply:
I wonder if there's a bit of a misunderstanding here. To clarify, checks list is actually results of the checks. I should probably rename it. So we check a thing and add the result to a list which would be returned to the caller.

Do you still have the same question?